paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -542,22 +543,20 @@ internal class PaymentSheetViewModel @Inject internal constructor(
                     _paymentSheetResult.tryEmit(PaymentSheetResult.Completed)
                 }
             }
-            else -> {
-                if (paymentResult is PaymentResult.Failed) {
-                    eventReporter.onPaymentFailure(
-                        paymentSelection = selection.value,
-                        currency = stripeIntent.currency,
-                        isDecoupling = isDecoupling,
-                    )
-                }
+            is PaymentResult.Failed -> {
+                eventReporter.onPaymentFailure(
+                    paymentSelection = selection.value,
+                    currency = stripeIntent.currency,
+                    isDecoupling = isDecoupling,
+                )
 
                 resetViewState(
-                    userErrorMessage = when (paymentResult) {
-                        is PaymentResult.Failed -> paymentResult.throwable.localizedMessage
-                        else -> null // indicates canceled payment
```

Comment:
Did you ensure that this is still valid?

Reply:
Yes, I did.